### DISCUSSION TITLE
Dropdown-list in register form.

### DISCUSSION THREAD

**Discussion by noahra** [keycloakify/keycloakify#458] [Q&A] [answered]
Hello! We would like to add a dropdown-list in our register form so users can input what country they come from. Is this possible?

**Comment by garronej** [comment, marked as answer]
Hello @noahra,   

Sure! Have a look at: https://docs.keycloakify.dev/realtime-input-validation

**Reply by garronej** [reply]
Look at the example with favourite_pet

**Reply by noahra** [reply]
That is perfect! But how can I toggle `User Profile Enabled`? I can't seem to find that option on 22.0.4.

**Reply by garronej** [reply]
See this notice:  
<img width="805" alt="image" src="https://github.com/keycloakify/keycloakify/assets/6702424/baa060f3-8e0d-4c7f-887c-2fcc21ed1c3c">

**Reply by noahra** [reply]
I successfully toggled the User Profile setting and added my Country-dropdown
![Screenshot 2023-11-30 at 14 11 24](https://github.com/keycloakify/keycloakify/assets/77496609/cce582d0-1ed9-4211-b1e1-aae913a9c9ff)

![Screenshot 2023-11-30 at 14 11 28](https://github.com/keycloakify/keycloakify/assets/77496609/c4b5cfa0-7f5a-4519-b932-d1b78be65e4c)



I added the mapper too.
![Screenshot 2023-11-30 at 14 23 17](https://github.com/keycloakify/keycloakify/assets/77496609/f1189a23-cbb5-43e6-b661-b7e26ec2d58d)


But the drop down does not show in the register page! what am I missing?

**Reply by garronej** [reply]
User can view and user can edit.  

<img width="1191" alt="image" src="https://github.com/keycloakify/keycloakify/assets/6702424/0c4eafa4-bc2e-4214-a394-fb395340ded9">

**Reply by noahra** [reply]
Thanks @garronej! This is good! Can you add options more efficiently? Like by writing code, in an infrastructure as code kind of way? Adding all countries manually is tedious and error-prone.

**Reply by garronej** [reply]
Sure you can import/export using JSON. 
![image](https://github.com/keycloakify/keycloakify/assets/6702424/6f26dd30-3c1b-4e31-aa09-43af5a520a84)

Beyond that I'd use GPT**4** to help me acheive a fully repoducible architecture as code setup.

**Reply by noahra** [reply]
Thanks a lot!!

**Reply by noahra** [reply]
is it possible to do a API Post with a JSON payload of countries?

**Reply by garronej** [reply]
You'd have to be more specific

**Reply by noahra** [reply]
Can I use the [Keycloak  Admin REST API](https://www.keycloak.org/docs-api/22.0.1/rest-api/index.html) to modify the dropdown list I create? For example, populate the dropdown with a list of alternatives?

**Reply by garronej** [reply]
Yes you can do whatever action you can do in the admin UI using the Admin REST API.  
You just have to open the network tab and see  what API endpoints you have to call to reproduce what you are doing by clicking button.  
Note however that calls to the Amin API can only be done from the backend. You can call this from your theme running in the browser. You can't ship the root user credential to the browser.  
You can however have an API that act as a proxy between your theme and the Keycloak Admin API.  

For example here, I get all the "organization of attach (bad translation from french)" of the other users so that I can provide suggestion as the user type and hopefully reduce the amount of users comming from the same organization but that used a slightly different spelling.  

https://github.com/keycloakify/keycloakify/assets/6702424/3dbc041b-f9df-4bb4-97e9-e5cd75d5a4c6
